common/scala/src/main/scala/whisk/core/entity/EntityPath.scala

```diff
@@ -189,13 +189,15 @@ protected[core] class EntityName private (val name: String) extends AnyVal {
 
 protected[core] object EntityName {
   protected[core] val ENTITY_NAME_MAX_LENGTH = 256
+  protected[core] val allowedSpecialCharacters = "@.&%!?=#-"
```

Comment:
really?!?

Reply:
all this valid/safe in URL? and CouchDB `_id` ?

Reply:
https://github.com/apache/incubator-openwhisk/issues/2830

rather than more arbitrary characters - since this isn't the first time the list has gotten larger, how about allowing the largest set of characters instead.

Reply:
proposing:

```
protected[core] val REGEX = raw"\A([a-zA-Z0-9]|[a-zA-Z0-9][^/]{0,${ENTITY_NAME_MAX_LENGTH - 2}}[^/ ])\z"
```

Will allow anything but "/" since that gets nasty in URLs.

Reply:
worth a dev list heads up.